std/string.d

```diff
@@ -3641,25 +3671,13 @@ in
 }
 body
 {
-    import std.exception : assumeUnique;
-
-    char[] transTable = new char[256];
+    immutable(char)[256] result = void;
```

Comment:
I don't understand why this can't just be `char[256]` and return as `immutable`. Will this break the NRVO?